cmd/registry/cmd/count/revisions.go

```diff
@@ -0,0 +1,154 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package count
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func revisionsCommand(ctx context.Context) *cobra.Command {
+	var filter string
+	cmd := &cobra.Command{
+		Use:   "revisions",
+		Short: "Count the number of revisions of specified resources",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			client, err := connection.NewClient(ctx)
+			if err != nil {
+				log.FromContext(ctx).WithError(err).Fatal("Failed to get client")
+			}
+			// Initialize task queue.
+			taskQueue, wait := core.WorkerPool(ctx, 64)
+			defer wait()
+			// Generate tasks.
+			name := args[0]
+			if spec, err := names.ParseSpec(name); err == nil {
+				err = core.ListSpecs(ctx, client, spec, filter, func(spec *rpc.ApiSpec) {
+					taskQueue <- &countSpecRevisionsTask{
+						client: client,
+						spec:   spec,
+					}
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to list API specs")
+				}
+			} else if deployment, err := names.ParseDeployment(name); err == nil {
+				err = core.ListDeployments(ctx, client, deployment, filter, func(deployment *rpc.ApiDeployment) {
+					taskQueue <- &countDeploymentRevisionsTask{
+						client:     client,
+						deployment: deployment,
+					}
+				})
+				if err != nil {
+					log.FromContext(ctx).WithError(err).Fatal("Failed to list API deployments")
+				}
+			} else {
+				log.FromContext(ctx).WithError(err).Fatal("Unsupported resource")
+			}
+		},
+	}
+	cmd.Flags().StringVar(&filter, "filter", "", "Filter selected resources")
+	return cmd
+}
+
+type countSpecRevisionsTask struct {
+	client connection.Client
+	spec   *rpc.ApiSpec
+}
+
+func (task *countSpecRevisionsTask) String() string {
+	return "count revisions " + task.spec.Name
+}
+
+func (task *countSpecRevisionsTask) Run(ctx context.Context) error {
+	count := 0
+	it := task.client.ListApiSpecRevisions(ctx,
+		&rpc.ListApiSpecRevisionsRequest{
+			Name: task.spec.Name,
+		})
+	for {
+		_, err := it.Next()
+		if err == iterator.Done {
+			break
+		} else if err == nil {
+			count++
+		} else {
+			return err
+		}
+	}
+	log.Debugf(ctx, "%d\t%s", count, task.spec.Name)
+	if task.spec.Labels == nil {
+		task.spec.Labels = make(map[string]string, 0)
+	}
+	task.spec.Labels["revisions"] = fmt.Sprintf("%d", count)
```

Comment:
Maybe `revision_count`?

Reply:
I considered that, and chose "revisions" for compact CEL expressions. I don't see much risk of ambiguity here since a representation of a collection of revisions wouldn't fit in a label.